Sources/Promise.swift

```
@@ -294,6 +294,20 @@ public struct Promise<Value> {
             }
         }
     }
+    
+    /**
+     Creates a promise that wraps this promise and performs a secondary task regardless of success or failure.
+     
+     - parameter resultTask: A function to be called that returns a promise.
+     - returns: A promise whose task runs this promise, and then calls the result task after it succeeds or fails.
+     */
+    public func always<U>(_ resultTask: @escaping () -> Promise<U>) -> Promise<U> {
+        return map { _ in () }.recover { _ in
+            Promise<()>(value: ())
+        }.flatMap { _ in
+            return resultTask()
+        }
+    }
```

This transformation does one more thing that your documentation doesn't mention, which is that it discards the result of the first task. And as a result, it means step two's input has to be void. That makes me think this is too specific, and we want more elementary and generic operations, like:

1. Map to (), which I don't think belongs in the library
2. Recover, but ignoring the error and assuming a default value (`recover(value:)`?), which in this case could be `()`
3. Do the second task (this is already just `flatMap`)

In other words I think you could write:

```
public func recover(value: Value) -> Promise<Value> {
    return recover { _ in
        Promise(value: value)
    }
}
```

… and then in any application, `taskOne.recover(value: someDefaultValue).flatMap(taskTwo)`.

In your application that would mean `taskOne.map { _ in }.recover(value: ()).flatMap(taskTwo)`.

---

@kconner this would solve my problem, but I don't think it makes enough of a difference that it is worth including. IMO I say we either update the documentation and potentially rename the `always` function or we just don't add it at all.